src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3110,9 +3110,9 @@ internal struct Buffer<TElement>
 
         internal Buffer(IEnumerable<TElement> source, bool queryInterfaces = true)
         {
-            TElement[] items = null;
-            int count = 0;
-
+            items = null;
+            count = 0;
```

Comment:
+1 here.
Keeping items and count as locals until the copying is done can be beneficial, so I'd preserve that. The change removes one null check (that would seem to be very predictable), so perf benefit of that might be very small and I wonder if introducing field accesses may actually make things slower.

* can we keep items/count as locals until done copying?
* what speedup are we looking at here overall?